src/core/control/tools/EditSelection.cpp

```diff
@@ -994,9 +1034,16 @@ void EditSelection::paint(cairo_t* cr, double zoom) {
     drawAnchorRect(cr, x + width, y + height, zoom);
 
 
-    drawDeleteRect(cr, std::min(x, x + width) - (DELETE_PADDING + this->btnWidth) / zoom, y, zoom);
+    drawAnchorDelete(cr, std::min(x, x + width) - (DELETE_PADDING + this->btnWidth) / zoom, y, zoom);
```

Comment:
Lots of duplication here. I think the code would be easier to read if some of the common computations were extracted out, e.g.
```
const double xmin = std::min(x, x + width);
const double xmax = std::max(x, x + width);
const double scaledBtnOffs = (DELETE_PADDING + this->btnWidth) / zoom;
```